sequencers/modeshape-sequencer-pdf/src/main/java/org/modeshape/sequencer/pdf/PdfBasicMetadata.java

```diff
@@ -0,0 +1,150 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modeshape.sequencer.pdf;
+
+import java.awt.print.PageFormat;
+import java.awt.print.PrinterException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Calendar;
+
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDDocumentInformation;
+import org.apache.pdfbox.pdmodel.PDPageable;
+
+/**
+ * Utility for extracting Document Information Directory metadata from PDF files.
+ */
+public class PdfBasicMetadata {
+
+    static final String MIME_TYPE_STRING = "application/pdf";
+
+    static final String[] ORIENTATION_STRINGS = {"landscape", "portrait", "reverse landscape"};
+
+    private Integer pageCount;
+    private String author;
+    private Calendar creationDate;
+    private String creator;
+    private String keywords;
+    private Calendar modificationDate;
+    private String producer;
+    private String subject;
+    private String title;
+    private String orientation;
+    private Boolean encrypted;
+    private String version;
+
+    private InputStream in;
+
+    /*
+     * Check that given file is supported by this sequencer.
+     */
+    public boolean check() {
+        try {
+            PDDocument document = PDDocument.load(in);
+            PDPageable pageable = new PDPageable(document);
+            PageFormat firstPage = pageable.getPageFormat(0);
+
+            encrypted = document.isEncrypted();
+            pageCount = document.getNumberOfPages();
+            orientation = ORIENTATION_STRINGS[firstPage.getOrientation()];
+            version = String.valueOf(document.getDocument().getVersion());
+            String catalogVersion = document.getDocumentCatalog().getVersion();
+            if (catalogVersion != null && !catalogVersion.isEmpty()) {
+                // According to specs version saved here should be determining instead
+                // the version in header. It is barely used, though.
+                version = catalogVersion;
+            }
+
+            if (!encrypted) {
+                PDDocumentInformation metadata = document.getDocumentInformation();
+                author = metadata.getAuthor();
+                creationDate = metadata.getCreationDate();
+                creator = metadata.getCreator();
+                keywords = metadata.getKeywords();
+                modificationDate = metadata.getModificationDate();
+                producer = metadata.getProducer();
+                subject = metadata.getSubject();
+                title = metadata.getTitle();
+
+                document.close();
+            }
+            return true;
+        } catch (IOException | IllegalArgumentException e) {
```

Comment:
these should either 
a) be logged here and therefore the `logger` object from the parent sequencer passed in and used here
b) left to bubble up and caught and logged by the sequencer

in either case we want to be aware that these exceptions occurred.